src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/ProjectSystem/RazorProjectInfoEndpointPublisher.cs

```diff
@@ -89,23 +91,14 @@ private void ProjectManager_Changed(object sender, ProjectChangeEventArgs args)
             case ProjectChangeKind.ProjectChanged:
             case ProjectChangeKind.DocumentRemoved:
             case ProjectChangeKind.DocumentAdded:
-                if (args.Newer is null)
-                {
-                    break;
-                }
-
-                EnqueuePublish(args.Newer);
+                EnqueuePublish(args.Newer.AssumeNotNull());
 
                 break;
 
             case ProjectChangeKind.ProjectAdded:
-
-                if (args.Newer is not null)
-                {
-                    // Don't enqueue project addition as those are unlikely to come through in large batches.
-                    // Also ensures that we won't get project removal go through the queue without project addition.
-                    ImmediatePublish(args.Newer, _disposeTokenSource.Token);
-                }
+                // Don't enqueue project addition as those are unlikely to come through in large batches.
+                // Also ensures that we won't get project removal go through the queue without project addition.
```

Comment:
I think the design of this should probably change to match the design of the ProjectStateSynchronizer in my PR, but I will need to update that PR after this merges anyway, so will do it there. Will also let me rename some RazorProjectService methods, which will be nice.

Reply:
This will happen in my next PR, which will remove the LSP notification completely.